### ISSUE TITLE
page.emulateMediaFeatures([{ name: 'hover', value: 'hover' }]) throws error

### ISSUE THREAD

**Issue by lehni** [puppeteer/puppeteer#5096]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 2.0.0, Headless
* Platform / OS version: Ubuntu 18.04.3 LTS
* URLs (if applicable):
* Node.js version: v10.16.3

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

1. Create a new page:
    ```js
    const page = await browser.newPage()
    ```
2. Emulate the hover media query:
    ```js
    await page.emulateMediaFeatures([{ name: 'hover', value: 'hover' }])
    ```

**What is the expected result?**

The feature should be emulated.

**What happens instead?**

The exception `Unsupported media feature: hover` is thrown

**Context**

When running Puppeteer headlessly on Linux, the result of the following media query is `false`:
```js
window.matchMedia('(hover: hover)').matches
```
Meanwhile on a Mac, the same query returns  `true`.

I would like to test media queries in our e2e tests, and not being able to control this behavior prevents this from being possible.

**Comment by ReDrUm** [comment]
Me too 👍
Support for `pointer` would be nice too e.g. `@media (hover: hover) and (pointer: fine)`

**Comment by denggj28** [comment]
after 3 years, the issue still exists :sweat:
@lehni do you have any solution?

**Comment by dthiessen-va** [comment]
Same issue here but with `width` and `device-width`.

> Unsupported media feature: width

> Unsupported media feature: device-width

**Comment by OrKoN** [comment]
I believe upstream those values are not supported and there is no plan to allow them https://bugs.chromium.org/p/chromium/issues/detail?id=1058326

**Comment by OrKoN** [comment]
If some of the values are actually supported by CDP, we can open them up in Puppeteer.

**Comment by Logesh08** [comment]
I found an alternative

```js
function preloadJs() {
    const originalMatchMedia = matchMedia;
    Object.defineProperty(window,'matchMedia',{
        value: function (querry) {
            class MediaQueryList{
                constructor(media,match){
                    this.media = media;
                    this.matches = match;
                    this.onChange = null;
                }
            }
            if(/(hover|pointer|color-gamut)/i.test(querry)){
                const params = querry.replaceAll("(",'').replaceAll(")",'').split(":").map(param=>param.trim());
                if(params.length===1){
                    const key = params[0];
                    if(/^(hover|any-hover|any-pointer|pointer|color-gamut)$/.test(key)){
                        return new MediaQueryList(querry,true);
                    }
                }
                if(params.length===2){
                    const value = params[1];
                    if(/^(fine|hover|srgb|p3)$/.test(value)){
                        return new MediaQueryList(querry,true);
                    }else{
                        return new MediaQueryList(querry,false);
                    }
                }
                
            }
            return originalMatchMedia(querry);    
        }
    })
}

await page.evaluateOnNewDocument(preloadJs));
```

I'm not sure if this is correct way of writing the code, but this actually solves the problem. 
I made it to bypass three CSS based tests 
1) hover or any-hover (true for hover)
2) pointer or any-pointer (true for fine and false for coarse)
3) color-gamut (true for srgb,p3 and false for rec2020)

Hope it helps someone.

**Comment by danyball** [comment]
@Logesh08 This will not fix media queries used in css like `@media (hover: hover)`, just if your own script uses `window.matchMedia`, right?

**Comment by Logesh08** [comment]
@danyball Yes you are right, it might help in js fingerprinting, but not in css fingerprinting. But I din't think about this one until you said.

**Comment by Kle0s** [comment]
I think the source of the issue here is that in linux the headless browser does not find a mouse in some cases, making it not support hover. This is not supported by emulation in CDP at the moment (@OrKoN, I assume this is still the case, but the old link is broken due to the ticket system migration).

**Comment by Logesh08** [comment]
Hey guys, I have fixed it.

There are switches which you can use for hover and pointer!!

**Comment by danyball** [comment]
Hey nice, thanks. In which version did your fix land?

**Comment by Logesh08** [comment]
Chrome 139 and puppeteer latest version

**Comment by danyball** [comment]
> Hey guys, I have fixed it.
> 
> There are switches which you can use for hover and pointer!!

Hey. Still wondering about this comment. What exactly do you mean by "switches for hover"? How to active it to not get the issue described in this issue ticket?

**Comment by Logesh08** [comment]
Hi, it's been a while. What I meant is the chrome launch arguments. By default the support to emulate hover was disabled in puppeteer and that was the reason of the issue described in the issue.

However I'm not sure about the latest version because I stopped using puppeteer long ago.
